presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergFileWriterFactory.java

```diff
@@ -134,6 +138,24 @@ private IcebergFileWriter createParquetWriter(
         List<String> fileColumnNames = icebergSchema.columns().stream()
                 .map(Types.NestedField::name)
                 .collect(toImmutableList());
+
+        List<Types.NestedField> icebergFields = new ArrayList<>(icebergSchema.columns());
+        ListIterator<Types.NestedField> icebergFieldsIterator = icebergFields.listIterator();
+
+        while (icebergFieldsIterator.hasNext()) {
+            Types.NestedField field = icebergFieldsIterator.next();
+            if (field.type() instanceof Types.TimestampType) {
```

Comment:
Do we need to consider the case for nested types? e.g. Array, Map, and Struct?

Reply:
Possibly, I'm going to build out the test first then see what the results are. If we do need to consider this we will probably need some kind of dfs style function